### ISSUE TITLE
Flag to ignore config schema violations

### ISSUE THREAD

**Issue by janosh** [astral-sh/ruff#5504]
Every time ruff adds new rules, my commit hooks start failing with

```sh
ruff.....................................................................Failed
- hook id: ruff
- exit code: 2

error: TOML parse error at line 68, column 3
   |
68 |   "PERF203", # try-except-in-loop
   |   ^^^^^^^^^
Unknown rule selector: `PERF203`
```

until [a PR to https://github.com/SchemaStore/schemastore](https://github.com/SchemaStore/schemastore/pulls?q=is%3Apr+Update+ruff+JSON++is%3Aclosed+) that updates the `ruff` schema is merged.

Since this seems hard to automate and sync with a ruff releases, maybe we could add a flag to `ruff check` to ignore schema violations?

**Related**
- https://github.com/astral-sh/ruff/issues/5321
- https://github.com/astral-sh/ruff/issues/4529

**Comment by charliermarsh** [comment]
Hmm, trying to understand why an update to the schema store is necessary here. What's enforcing that the schema matches that of the store? Do you have another pre-commit hook that's enforcing schema validity?

To put it differently: is it Ruff that's throwing an error here? If so, it seems like the Ruff version used by your pre-commit hook doesn't match that of your configuration file.

**Comment by janosh** [comment]
Oops I thought I bumped the ruff version in `.pre-commit-config.yaml` but maybe I didn't save correctly. Was still at outdated `ruff==0.0.275`.

**Comment by charliermarsh** [comment]
Okay cool, that makes more sense! Thanks for following up.

**Comment by charliermarsh** [comment]
That did, however, remind me to update our JSON schema: https://github.com/SchemaStore/schemastore/pull/3048.
